### PR TITLE
Allow pragma to be set by env variable

### PR DESCRIPTION
**Pull request by navjotk**
When using the `CustomCompiler`, the pragma used for ignoring vector dependencies is the one for the intel compiler. This change allows the pragma to be set with an environment variable.
This continues the thought behind having a compiler completely configurable by environment variables.

### CODE REVIEW

File: devito/compiler.py [hunk 1]
```diff
@@ -192,7 +192,7 @@ def __init__(self, *args, **kwargs):
         default = '-O3 -g -march=native -fPIC -mno-avx -Wall -std=c99'
         self.cflags = environ.get('CFLAGS', default).split(' ')
         self.ldflags = environ.get('LDFLAGS', '-shared').split(' ')
-
+        self.pragma_ivdep = [Pragma(environ.get('DEVITO_PRAGMA', 'GCC ivdep'))]
```
Comment (mlange05) [inline devito/compiler.py]: Shouldn't the env name be slightly more specific? Maybe `DEVITO_IVDEP` or `DEVITO_PRAGMA_IVDEP`?

Comment (navjotk) [inline devito/compiler.py]: I reiterate this sentiment on the use of `t` vs `time` in the entire codebase :)

**Comment by FabioLuporini** [comment]
OK

does this impact StencilKernel too?i wouldn't think so but maybe Im wrong

File: devito/compiler.py (hunk 1 above)
Comment (mlange05) [inline devito/compiler.py]: I'm not sure the above comment was meant to be made here... ?

**Comment by mlange05** [comment]
Compiler presets are agnostic to the operator classes, so yes, this will affect `StencilKernel` as well.

**Comment by mlange05** [comment]
Good to go, so merging...

**Comment by FabioLuporini** [comment]
> Compiler presets are agnostic to the operator classes, so yes, this will affect StencilKernel as well.

Sure, but I was just wondering whether that `.pragma_ivdep` was even used in 3.0, as I don't remember I needed it. Anyway, fine.

**Pull request merged**